host/tests/dsClasses/dsHdmiIn.py

```diff
@@ -0,0 +1,445 @@
+                    "query": "Select dsAudio Port Index[0-10]:",
+                    "input": str(port_index)
+                },
+                {
+                    "query_type": "direct",
+                    "query": "Select audiomix value Index[0-1]:",
+                    "input": str(audmix)
+                },
+                {
+                    "query_type": "direct",
+                    "query": "Select videoplane value Index[0]:",
+                    "input": str(videoplane)
+                },
+                {
+                    "query_type": "direct",
+                    "query": "Select videoplane value Index[1]:",
+                    "input": str(topmost)
+                }
+        ]
+
+        result = self.utMenu.select(self.testSuite, "Select Port", promptWithAnswers)
+
+    def scalevdieo(self, hdmiin_port:str, xcord:int=0, ycord:int=0, width:int=0, height:int=0):
+        """
+        Scale video of specified port.
+
+        Args:
+            xcord (int, optional): xcoordiante. Defaults to 0
+            ycord (int, optional): ycoordiante. Defaults to 0
+            width (int, optional): width. Defaults to 0
+            height(int, optional): height. Defaults to 0
+
+        Returns:
+            None
+        """
+        promptWithAnswers = [
+                {
+                    "query_type": "list",
+                    "query": "Select dsAudio Port:",
+                    "input": hdmiin_port
+                },
+                {
+                    "query_type": "direct",
+                    "query": "Select dsAudio Port Index[0-10]:",
+                    "input": str(xcord)
+                },
+                {
+                    "query_type": "direct",
+                    "query": "Select audiomix value Index[0-1]:",
+                    "input": str(ycord)
+                },
+                {
+                    "query_type": "direct",
+                    "query": "Select videoplane value Index[0]:",
+                    "input": str(width)
+                },
+                {
+                    "query_type": "direct",
+                    "query": "Select videoplane value Index[1]:",
+                    "input": str(height)
+                }
+        ]
+
+        result = self.utMenu.select(self.testSuite, "Scale Video", promptWithAnswers)
+  
+    def getSupportedPorts(self):
+        """
+        Returns a list of supported audio ports on the device.
+
+        Args:
+            None
+
+        Returns:
+            list: A list of tuples containing the port name and index.
+        """
+
+        ports = self.deviceProfile.get("Ports")
+        if not ports:
+            return []  # Handle empty ports list
+
+        supported_ports = []
+        for entry in ports.values():
+            supported_ports.append((dsHdmiInPortType(entry['Typeid']).name, entry['Index']))
+
+        return supported_ports
+
+    def zoommode(zoom_mode:int=0):
```

Comment:
rename the function name (eg: setZoomMode)

Reply:
done